Marlin/Marlin_main.cpp

```diff
@@ -5038,7 +5038,7 @@ inline void gcode_M121() { endstops.enable_globally(false); }
    */
   inline void gcode_M156() {
     uint8_t addr = code_seen('A') ? code_value_short() : 0;
-    int bytes    = code_seen('B') ? code_value_short() : 0;
+    int bytes    = code_seen('B') ? code_value_long() : 0;
 
     if (addr && bytes) {
       i2c.address(addr);
```

Comment:
Ah, very good. It seemed to me that somewhere in time and space there might be a request for more than 65,535 bytes over an i2c bus, but in hindsight, quite right, that would be a bit much.